### PR TITLE
WIP - feat/kadena-cli/tx/sign

### PR DESCRIPTION
**Pull request by nillo**

### CODE REVIEW

File: packages/tools/kadena-cli/src/tx/commands/txCreateTransaction.ts [hunk 1]
```diff
@@ -66,40 +66,45 @@ export const createTransactionCommandNew = createCommandFlexible(
...
+      console.log(result.data.transaction);
 
-    const relativePath = path.relative(process.cwd(), result.data.filePath);
-    console.log(`\ntransaction saved to: ./${relativePath}`);
+      const relativePath = path.relative(process.cwd(), result.data.filePath);
+      console.log(`\ntransaction saved to: ./${relativePath}`);
 
-    return { outFile: relativePath };
+      return { outFile: relativePath };
+    } catch (error) {
```
Comment (barthuijgen) [inline packages/tools/kadena-cli/src/tx/commands/txCreateTransaction.ts:105]: we should probably just do this try{}catch{} in the createCommand right? if its such a generic handling

Comment (KdWarlordz) [reply packages/tools/kadena-cli/src/tx/commands/txCreateTransaction.ts:105]: would be cleaner yeah,

File: packages/tools/kadena-cli/src/tx/utils/txHelpers.ts [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+ * @param {boolean} signed - Whether to retrieve signed or unsigned transactions.
+ * @returns {Promise<string[]>} A promise that resolves to an array of transaction file names.
+ * @throws Throws an error if reading the transaction directory fails.
+ */
+export async function getTransactions(
+  signed: boolean,
+  path: string,
+): Promise<string[]> {
+  try {
+    const filePath = path ? `${process.cwd()}/${path}` : TRANSACTION_PATH;
```
Comment (barthuijgen) [inline packages/tools/kadena-cli/src/tx/utils/txHelpers.ts:67]: Prefer using path.join

File: packages/tools/kadena-cli/src/tx/utils/txHelpers.ts [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+        : join(TRANSACTION_PATH, transactionFile);
+    const transactionFilePath = join(TRANSACTION_PATH, transactionFile);
+    const fileContent = await services.filesystem.readFile(filePath);
+
+    if (fileContent === null) {
+      throw Error(`Failed to read file at path: ${transactionFilePath}`);
+    }
+    const transaction = JSON.parse(fileContent);
+    if (signed) {
+      tx.ICommandSchema.parse(transaction);
```
Comment (barthuijgen) [inline packages/tools/kadena-cli/src/tx/utils/txHelpers.ts:248]: Parse should also return the data with correct type so you don't need to use "as x"

**Review by barthuijgen — APPROVED** [review]

**Pull request merged**